c++/nda/_impl_basic_array_view_common.hpp

```diff
@@ -262,6 +262,41 @@ FORCEINLINE decltype(auto) operator()(Ts const &...idxs) && noexcept(has_no_boun
   return call<Algebra, true>(*this, idxs...);
 }
 
+private:
+// Right now we are only doing SIMD access in contiguous layouts. If this rule is relaxed we need to change this function as well.
+void assert_simd_access_bounds(const long offset) const noexcept(has_no_boundcheck) {
+  static_assert(
+     has_contiguous_layout<self_t>,
+     "This functions should only be called when we have a contiguous layout. This can fail only when the rules of vectorization is relaxed therefore this function needs to be updated");
+  if constexpr (!has_no_boundcheck) { assert(offset + native_simd<ValueType>::size <= this->size() && "NDA: SIMD access out of bounds"); }
```

Comment:
Assertions are not exceptions, so you can drop all the `noexcept` specifiers (which will also generate more efficient code, because the compiler will not have to emit instructions to check whether an exception was thrown).

There is also an [`ASSERT_WITH_MESSAGE`](https://github.com/TRIQS/nda/blob/unstable/c%2B%2B/nda/macros.hpp#L76) macro.
```suggestion
  if constexpr (!has_no_boundcheck) { ASSERT_WITH_MESSAGE(offset + native_simd<ValueType>::size <= this->size(), "NDA: SIMD access out of bounds") }
```

Reply:
I replaced assert with runtime error. That was how it was done for bound checking for scalar access.